getgauge/util.py

```diff
@@ -14,3 +14,19 @@ def get_project_root():
 
 def get_step_impl_dir():
     return os.path.join(get_project_root(), STEP_IMPL_DIR_NAME)
+
+def list_impl_files(file_list,step_impl_dir = get_step_impl_dir()):
+    for f in os.listdir(step_impl_dir):
```

Comment:
use `os.walk` instead of `os.listdir`.
use a list comprehension instead `for` and `recursion` combination